core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -1832,15 +1832,21 @@ class WorkspaceService(protected val userInfo: UserInfo, val dataSource: SlickDa
   }
 
   def checkSamActionWithLock(workspaceName: WorkspaceName, samAction: SamResourceAction): Future[PerRequestMessage] = {
-    val testFuture = dataSource.inTransaction { dataAccess =>
-      withWorkspaceContext(workspaceName, dataAccess, Some(WorkspaceAttributeSpecs(all=false))) { workspaceContext =>
-        requireAccess(workspaceContext.workspace, samAction) {
-          DBIO.successful(RequestComplete(StatusCodes.NoContent)) //if we get here, we passed all the hoops
-        }
+    val wsCtxFuture = dataSource.inTransaction { dataAccess =>
+      withWorkspaceContext(workspaceName, dataAccess, Some(WorkspaceAttributeSpecs(all = false))) { workspaceContext =>
+        DBIO.successful(workspaceContext)
+      }
+    }
```

Comment:
NOT for this PR - just for saying it out loud: `withWorkspaceContext` itself contains REST calls to Sam, and is within the transaction. I don't wanna blow up scope too much so I'm ok with it - just wanna get it out there in case folks don't realize that.

Reply:
> `withWorkspaceContext` itself contains REST calls to Sam

I don't see any?

Reply:
ah dammit I should maybe just go back to bed. I was thinking of `withWorkspaceContextAndPermissions`. My fault   here - sorry